src/components/WalletUnlock/DeterministicAccountList.tsx

```diff
@@ -88,87 +80,120 @@ const SDownloader = styled(Downloader)`
   }
 `;
 
+export const filterZeroBalanceAccounts = (accounts: DWAccountDisplay[]) =>
+  accounts.filter(({ balance }) => balance?.isZero());
+
 interface DeterministicAccountListProps {
   finishedAccounts: DWAccountDisplay[];
   asset: ExtendedAsset;
   isComplete: boolean;
   network: Network;
-  freshAddressIndex: number;
-  generateFreshAddress(): void;
+  displayEmptyAddresses: boolean;
+  selectedDPath: DPath;
+  handleScanMoreAddresses(dpath: ExtendedDPath): void;
   onUnlock(param: any): void;
   handleUpdate(asset: ExtendedAsset): void;
 }
 
-interface ISelectedAccount {
-  address: TAddress;
-  derivationPath: string;
-}
+const tableBannerText = (emptySelectedAccountsLength: number, maxEmptyAddresses: number) => (
+  <Trans
+    id="DETERMINISTIC_SCANNING_EMPTY_ADDR"
+    variables={{
+      $count: () => emptySelectedAccountsLength,
+      $total: () => maxEmptyAddresses
+    }}
+  />
+);
 
 export default function DeterministicAccountList({
   finishedAccounts,
   asset,
   isComplete,
-  onUnlock,
   network,
-  freshAddressIndex,
-  generateFreshAddress,
+  displayEmptyAddresses,
+  selectedDPath,
+  handleScanMoreAddresses,
+  onUnlock,
   handleUpdate
 }: DeterministicAccountListProps) {
+  const MAX_EMPTY_ADDRESSES = 5;
```

Comment:
Perhaps this should be defined more globally and imported?

Reply:
ehhh its only used in this component, so i'm not sure what the point of moving it to our global configs would be.